src/libcore/iter/traits.rs

```diff
@@ -901,6 +901,10 @@ impl<I, T, E> Iterator for ResultShunt<I, E>
             None => None,
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.iter.size_hint()
```

Comment:
This looks incorrect since this iterator can yield fewer items than `self.iter`. Also this particular iterator is private and as far as can tell its `size_hint` is never used.

Reply:
Yes, this should instead be:
```
let (_, upper) = self.iter.size_hint();
(0, upper)
```
And I guess it's never used? I didn't really look; figuring out if it's used is harder than implementing it just-in-case, and the code around it could change.

I can remove it if you like.

Reply:
I *think* it’s not used but this implementation is small enough that it’s probably fine to add it just in case it becomes used later.